numpy/core/src/umath/ufunc_object.c

```diff
@@ -3970,17 +3971,26 @@ PyUFunc_GenericReduction(PyUFuncObject *ufunc, PyObject *args,
         }
     }
     else {
-        if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|OO&O&i:reduce",
+        if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|OO&Oi:reduce",
                                         reduce_kwlist,
                                         &op,
                                         &axes_in,
                                         PyArray_DescrConverter2, &otype,
-                                        PyArray_OutputConverter, &out,
+                                        &out_obj,
                                         &keepdims)) {
             Py_XDECREF(otype);
             return NULL;
         }
     }
+    /* if out_obj is a tuple with size 1 (=nout here), unpack it */
+    if (out_obj != NULL && PyTuple_CheckExact(out_obj) &&
+            PyTuple_GET_SIZE(out_obj) == 1) {
+        out_obj = PyTuple_GET_ITEM(out_obj, 0);
+    }
```

Comment:
OK, done, except a `TypeError`, as in `override.c`, because the situation here seems most analogous to not given sufficient number of arguments to a function, in which case cpython also raises a `TypeError` (see first item of the post-merger to-do list in #8892).